### PR TITLE
use the plain byte alphabet of SDSL submodule

### LINKED ISSUE #1603: Move `csa_alphabet_strategy.hpp` to SDSL
- [x] copy to SDSL
- [x] add tests to SDSL
- [x] add documentation in SDSL (Muhahahahaha)
- [x] remove seqan version
- [x] run test suit to check if everything works
- [x] update submodule to point to latest head if sdsl PR 
merged

### PR DESCRIPTION
**Pull request by joergi-w**
Co-authored-by: Lydia Buntrock <lydia.buntrock@fu-berlin.de>

We updated the sdsl, such that it contains the plain byte alphabet and removed it from the seqan3 repository. This PR updates the SDSL submodule.

Resolves #1603 

Includes PR https://github.com/xxsds/sdsl-lite/pull/72

### CODE REVIEW

File: include/seqan3/search/fm_index/fm_index.hpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 #pragma once
 
+#include <sdsl/csa_alphabet_strategy.hpp>
```
Comment (eseiler) [inline include/seqan3/search/fm_index/fm_index.hpp]: This should not be necessary

File: include/seqan3/search/fm_index/fm_index_cursor.hpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include <array>
 #include <type_traits>
 
+#include <sdsl/csa_alphabet_strategy.hpp>
```
Comment (eseiler) [inline include/seqan3/search/fm_index/fm_index_cursor.hpp]: This should not be necessary

**Comment by smehringer** [comment]
@joergi-w  @Irallia 
Can you reference the PR of the SDSL that include the byte alphabet here? Then I'll merge

**Comment by joergi-w** [comment]
This PR includes https://github.com/xxsds/sdsl-lite/pull/72 of the SDSL.

**Pull request merged**